lib/active_record/tenanted/cross_tenant_associations.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Tenanted
+    module CrossTenantAssociations
+      extend ActiveSupport::Concern
+
+      class_methods do
+        # I think we can have more configs later
+        def cross_tenant_config(**config)
+          @cross_tenant_config = config
+        end
+
+        def get_cross_tenant_config
+          @cross_tenant_config ||= {}
+        end
+
+        def has_one(name, scope = nil, **options)
+          define_enhanced_association(:has_one, name, scope, **options)
+        end
+
+        def has_many(name, scope = nil, **options)
+          define_enhanced_association(:has_many, name, scope, **options)
+        end
+
+        private
+          # For now association methods are identical
+          def define_enhanced_association(association_type, name, scope, **options)
+            config = get_cross_tenant_config
+            tenant_column = config[:tenant_column] || :tenant_id
```

Comment:
I don't want to require a tenant column in the table. The implementation must be able to sort this all out without relying on model attributes!